doc/config.rst

```diff
@@ -353,6 +353,39 @@ General configuration
 
    .. versionadded:: 1.3
 
+.. confval:: smart_quotes
```

Comment:
I agree the basic concept of this PR. But we would need to be able to disable smart_quotes by several condition.
For example, #4142 tells us to disable it for Japanese document. I saw somebody wants to disable it for arbitrary builder in some issues.
So I think this should not be boolean. (but I don't have good alternative idea...)

Reply:
I have addressed that by supplementary `smartquotes_excludes` dict configuration setting at 88a6e2451 (fixed typo in doc at 4b3a09133)